### PR TITLE
Commit full client context changes

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EXUI-1838
https://tools.hmcts.net/jira/browse/EXUI-1839
https://tools.hmcts.net/jira/browse/EXUI-1840

### Change description ###
Replace take to be completed in session storage with new client context object

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by RiteshHMCTS — DISMISSED** [review]

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Looks good apart from one v minor typo

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/services/cases.service.spec.ts [hunk 1]
```diff
@@ -389,9 +404,10 @@ describe('CasesService', () => {
       const headers = new HttpHeaders()
         .set('experimental', 'true')
         .set('Accept', CasesService.V2_MEDIATYPE_CASE_DATA_VALIDATE)
-        .set('Content-Type', 'application/json');
+        .set('Content-Type', 'application/json')
+        .set('Client-Context', window.btoa(JSON.stringify(CLIENT_CONTEXT)));
 
-      expect(httpService.post).toHaveBeenCalledWith(VALIDATE_CASE_URL, CASE_EVENT_DATA, { headers, observe: 'body' });
+      expect(httpService.post).toHaveBeenCalledWith(VALIDATE_CASE_URL, CASE_EVENT_DATA, { headers, observe: 'response' as 'bodu' });
```
[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/services/cases.service.spec.ts]: Edited, great spot

**Review by andywilkinshmcts — DISMISSED** [review]

**Review by Josh-HMCTS — DISMISSED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/services/event-completion-state-machine.service.ts [hunk 2]
```diff
@@ -216,4 +216,11 @@ export class EventCompletionStateMachineService {
       this.stateFinal
     );
   }
+
+  public taskPresentInSessionStorage(context: EventCompletionStateMachineContext): boolean {
+    const clientContextStr = context.sessionStorageService.getItem('clientContext');
+    console.log('clienht cintedt ', clientContextStr)
```
[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/services/event-completion-state-machine.service.ts:222]: will remove

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/case-edit/case-edit.component.ts [hunk 3]
```diff
@@ -475,14 +474,16 @@ export class CaseEditComponent implements OnInit, OnDestroy {
...
-    if (taskStr && assignNeeded) {
-      const task: Task = JSON.parse(taskStr);
+    if (task && assignNeeded && taskToBeCompleted) {
+      // const task: Task = JSON.parse(taskStr);
       this.abstractConfig.logMessage(`postCompleteTaskIfRequired with assignNeeded: taskId ${task.id} and event name ${this.eventTrigger.name}`);
       return this.workAllocationService.assignAndCompleteTask(task.id, this.eventTrigger.name);
-    } else if (taskStr) {
-      const task: Task = JSON.parse(taskStr);
+    } else if (task && taskToBeCompleted) {
+      // const task: Task = JSON.parse(taskStr);
```
[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/case-edit/case-edit.component.ts:486]: yes we can

**Review by RiteshHMCTS — APPROVED** [review]

**Review by Josh-HMCTS — APPROVED** [review]

**Pull request merged**